pixels-cli/src/main/java/io/pixelsdb/pixels/cli/load/PixelsConsumer.java

```diff
@@ -244,21 +307,35 @@ public void run()
 
     /**
      * Close the pixels writer and add the file to loaded file queue.
-     * Files in the loaded files queue will be saved in metadata.
+     * Files in the loaded files queue will be updated in metadata.
      * @param pixelsWriter the pixels writer
-     * @param fileName the file name without directory path
+     * @param loadedFile the file name has been loaded
      * @param filePath the path of the directory where the file was written
      * @throws IOException
      */
-    private void closeWriterAndAddFile(PixelsWriter pixelsWriter, String fileName, Path filePath) throws IOException
+    private void closeWriterAndAddFile(PixelsWriter pixelsWriter, File loadedFile, Path filePath) throws IOException
     {
         pixelsWriter.close();
-        File loadedFile = new File();
-        loadedFile.setName(fileName);
         loadedFile.setType(File.Type.REGULAR);
         loadedFile.setNumRowGroup(pixelsWriter.getNumRowGroup());
-        loadedFile.setPathId(filePath.getId());
         this.loadedFiles.offer(loadedFile);
         this.loadedPaths.offer(filePath);
     }
+
+    /**
+     * Create a temporary file through the metadata service
+     * @param fileName the file name without directory path
+     * @param filePath the path of the directory where the file was written
+     */
+    private File openTmpFile(String fileName, Path filePath) throws MetadataException
```

Comment:
Why create a tmp file? We use a tmp empty file in retina for the write buffer. However, for the load command, it is fine to directly write the file to the path and add it to the metadata.